fip-0025.md

```diff
@@ -0,0 +1,80 @@
+---
+fip: 25
+title: Return bundle transaction count in get_fio_names
+status: Draft
+type: Functionality
+author: Pawel Mastalerz <pawel@fioprotocol.io>
+created: 2021-01-19
+updated: 2021-01-19
+---
+
+# Abstract
+This FIP modifies get_fio_names to return remaining bundled transaction count.
+
+Modified getters:
+|Contract|Action|Endpoint|Description|
+|---|---|---|---|
+|||get_fio_names|Modified to add remaining bundled transaction count.|
+
+# Motivation
+It is currently difficult to obtain remaining bundled transaction count for a FIO Address. It can be fetched using [get_table_rows](https://developers.fioprotocol.io/api/api-spec/reference/get-table-rows/get-table-rows), but it requires converting the FIO Address to a hash which is the last 16 bytes of a sha1 hash (of the string), as big-endian hex.
+
+# Specification
+### Modifications to existing API calls
+#### [/get_fio_names](https://developers.fioprotocol.io/api/api-spec/reference/get-fio-names/get-fio-names)
+Modified to add remaining bundled transaction count.
+##### Request
+Unchanged
+###### Example
+```
+{
+	"fio_public_key": "FIO8PRe4WRZJj5mkem6qVGKyvNFgPsNnjNN6kPhh6EaCpzCVin5Jj"
+}
+```
+##### Processing
+Unchanged
+##### Exception handling
+Unchanged
+##### Response
+|Group|Parameter|Format|Definition|
+|---|---|---|---|
+|fio_domains|fio_domain|String|FIO Domain.|
+|fio_domains|expiration|String|Date when domain expires.|
+|fio_domains|is_public|Int|0 - domain is not public; 1 - domain is public.|
+|fio_addresses|fio_address|String|FIO Address.|
+|fio_addresses|expiration|String|Date when address expires.|
+|fio_addresses|remaining_bundled_tx|Int|Count of remaining bundled transactions.|
+###### Example
+```
+{
+  "fio_domains": [
+    {
+      "fio_domain": "alice",
+      "expiration": "2020-09-11T18:30:56",
+      "is_public": 0
+    }
+  ],
+  "fio_addresses": [
+    {
+      "fio_address": "purse@alice",
+      "expiration": "2020-09-11T18:30:56",
+      "remaining_bundled_tx": 98
+    }
+  ]
+}
+```
+
+# Rationale
+Modification to existing getter is deemed to be the best approach.
+
+# Implementation
+TBD
+
+# Backwards Compatibility
+[/get_fio_names](https://developers.fioprotocol.io/api/api-spec/reference/get-fio-names/get-fio-names) is the only existing API method being modified, but only a new response element is added, so wallets currently implementing this call should not be affected.
```

Comment:
> only a new response element is added, so wallets currently implementing this call should not be affected

Is this true for integrations that might serialize the responses into objects? Have we decided adding new fields to response objects is a safe, backward compatible operation?

Reply:
That has been our approach. Last in FIP-6.